pot/src/ser.rs

```diff
@@ -649,40 +651,36 @@ impl SymbolMap {
         // address of the str in the map.
         let symbol_address = symbol.as_ptr() as usize;
         // Perform a binary search to find this existing element.
-        match self
-            .static_lookup
-            .binary_search_by(|check| symbol_address.cmp(&check.0))
-        {
-            Ok(position) => RegisteredSymbol {
-                id: self.static_lookup[position].1,
+        match self.static_lookup.entry(symbol_address) {
+            indexmap::map::Entry::Occupied(position) => RegisteredSymbol {
+                id: *position.get(),
                 new: false,
             },
-            Err(position) => {
+            indexmap::map::Entry::Vacant(position) => {
                 // This static symbol hasn't been encountered before.
-                let symbol = self.find_entry_by_str(symbol);
-                self.static_lookup
-                    .insert(position, (symbol_address, symbol.id));
+                let symbol = Self::find_entry_by_str(&mut self.entries, &mut self.symbols, symbol);
+                position.insert(symbol.id);
                 symbol
             }
         }
     }
 
     #[allow(clippy::cast_possible_truncation)]
-    fn find_entry_by_str(&mut self, symbol: &str) -> RegisteredSymbol {
-        match self
-            .entries
-            .binary_search_by(|check| self.symbols[check.0.clone()].cmp(symbol))
-        {
+    fn find_entry_by_str(
+        entries: &mut Vec<(Range<usize>, u32)>,
+        symbols: &mut String,
+        symbol: &str,
+    ) -> RegisteredSymbol {
+        match entries.binary_search_by(|check| symbols[check.0.clone()].cmp(symbol)) {
             Ok(index) => RegisteredSymbol {
-                id: self.entries[index].1,
+                id: entries[index].1,
                 new: false,
             },
             Err(insert_at) => {
-                let id = self.entries.len() as u32;
-                let start = self.symbols.len();
-                self.symbols.push_str(symbol);
-                self.entries
-                    .insert(insert_at, (start..self.symbols.len(), id));
+                let id = entries.len() as u32;
+                let start = symbols.len();
+                symbols.push_str(symbol);
+                entries.insert(insert_at, (start..symbols.len(), id));
```

Comment:
wont you have the same 'insert' concern here?  whenever we insert a new static string via `find_or_insert` we should also be inserting a new string here (since I'm pretty sure the rust compiler deduplicates string constants).